br/pkg/storage/s3.go

```diff
@@ -358,38 +359,45 @@ func NewS3Storage(ctx context.Context, backend *backuppb.S3, opts *ExternalStora
 		)
 	}
 	c := s3.New(ses, s3CliConfigs...)
-	confCred := ses.Config.Credentials
-	setCredOpt := func(req *request.Request) {
-		// s3manager.GetBucketRegionWithClient will set credential anonymous, which works with s3.
-		// we need reassign credential to be compatible with minio authentication.
-		if confCred != nil {
-			req.Config.Credentials = confCred
+
+	// s3manager.GetBucketRegionWithClient failed to get region from client if the client is ceph or minio
+	// It defaut back to us-west-1 which break the code if a region is already set
+	if qs.Provider != "ceph" && qs.Provider != "minio" {
```

Comment:
1. Not sure if excluding ceph and minio is the right solution, maybe we should check if `qs.Provider == "s3" ` (or GCS?)
2. Maybe we should separate the provider config from the logic here. And check if `ProviderConfig.check_bucket_with_region == true` and then elsewhere store the config for each provider.

These are just ideas, I have no objections on merging this as-is.

Reply:
1 . I'm not sure about the list. We add ceph and minio because it's what we are working with and had troubles, but the issues may be wider.
2 Would it be simpler to add a parameter like qs.SkipBucketRegionCheck and a tag --skip-bucket-region-check when we launch br ? ( the wording is a bit long ^^")